console/src/data/ServiceLayer.js

```diff
@@ -44,6 +44,7 @@ const ServiceLayerBase = hoc((configHoc, Wrapped) => {
 			this.maps = new Set();
 			this.location = this.props.location;
 			this.isFirstPosition = true;
+			this.destinationReached = false;
```

Comment:
Should this be state rather than an instance member?

Reply:
It didn't seem like it needed to be. I didn't need to invoke a render when setting it. The popup is controlled through an appState prop. This is just an internal tracker. Maybe you're seeing or thinking of a scenario I wasn't considering?

Reply:
Ideally, you only have one source of truth for data so it doesn't get out of sync. In a react app, that's further complicated by batched state updates. I don't have a specific scenario; just a general observation.

Reply:
There is only one source of truth. The other prop I was referring to is controlling something else somewhere else in the app. I'm open to suggestions on how to improve this though, but you'll have to explain how setState will be better than an instance variable for this occasion because I'm not following that part.

Reply:
Instance variables open the risk of data managed by `setState` becoming out of sync with instance variables because they are updated immediately and `setState` can be batched. If that doesn't apply here, then you're fine.

Reply:
It doesn't. The appState and instance variables are not meant to be in sync. The instance variable is "whether we've done the initial triggering of the destination reached appState for the given destination" while the appState is "whether the popup is presently onscreen". Once the popup is dismissed (and removed from appState) we don't want it showing up again for the given destination. E.g. The car is still moving and we report "Destination Reached" popup at 10 meters from target. If it's dismissed at 9m away, and the car still moves toward the true goal, we don't want another popup displaying again; at least until a new destination is chosen.